src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import React, { useState } from 'react';
+
 import './App.scss';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getNewTodoId, getUserById } from './helper';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const invalidSymbols = /[^a-zA-Za-åa-ö-w-я 0-9/ .,]/gi;
+    const normalizedValue = event.target.value.replaceAll(invalidSymbols, '');
```

Comment:
it's a good practice to move such functions to helpers (outside handleTitleChange and App)